x-pack/legacy/plugins/siem/public/pages/detection_engine/components/signals/actions.tsx

```diff
@@ -58,19 +74,117 @@ export const updateSignalStatusAction = async ({
 
-export const sendSignalsToTimelineAction = async ({
+export const sendSignalToTimelineAction = async ({
+  apolloClient,
   createTimeline,
-  data,
-}: SendSignalsToTimelineActionProps) => {
-  const stringFilter = data[0].filter(d => d.field === 'signal.rule.filters')?.[0]?.value ?? [];
+  ecsData,
+  updateTimelineIsLoading,
+}: SendSignalToTimelineActionProps) => {
+  const timelineId =
+    ecsData.signal?.rule?.timeline_id != null ? ecsData.signal?.rule?.timeline_id[0] : '';
 
-  // TODO: Switch to using from/to when adding dateRange
-  // const [stringFilters, from, to] = getFilterAndRuleBounds(data);
-  const parsedFilter = stringFilter.map(sf => JSON.parse(sf));
-  createTimeline({
-    id: 'timeline-1',
-    filters: parsedFilter,
-    dateRange: undefined, // TODO
-    kqlQuery: undefined, // TODO
-  });
+  const ellapsedTimeRule = (ecsData.signal?.rule?.from != null
+    ? ecsData.signal?.rule?.from[0]
+    : 'now-0s'
+  )
+    .replace('now-', '')
+    .replace('s', '');
+
+  const from = moment(ecsData.timestamp ?? new Date())
+    .subtract(ellapsedTimeRule, 's')
+    .valueOf();
+  const to = moment(ecsData.timestamp ?? new Date()).valueOf();
+
+  if (timelineId !== '' && apolloClient != null) {
+    try {
+      updateTimelineIsLoading({ id: 'timeline-1', isLoading: true });
+      const responseTimeline = await apolloClient.query<
+        GetOneTimeline.Query,
+        GetOneTimeline.Variables
+      >({
+        query: oneTimelineQuery,
+        fetchPolicy: 'no-cache',
+        variables: {
+          id: timelineId,
+        },
+      });
+
+      const timelineTemplate: TimelineResult = omitTypenameInTimeline(
+        getOr({}, 'data.getOneTimeline', responseTimeline)
+      );
+      const { timeline } = formatTimelineResultToModel(timelineTemplate, true);
+      const query = replaceTemplateFieldFromQuery(
+        timeline.kqlQuery?.filterQuery?.kuery?.expression ?? '',
+        ecsData
+      );
+      const filters = replaceTemplateFieldFromMatchFilters(
+        (timeline.filters ?? []) as esFilters.Filter[],
+        ecsData
+      );
+      const dataProviders = replaceTemplateFieldFromDataProviders(
+        (timeline.dataProviders ?? []) as DataProvider[],
+        ecsData
+      );
+      createTimeline({
+        from,
+        timeline: {
+          ...timeline,
+          dataProviders,
+          eventType: 'all',
+          filters,
+          dateRange: {
+            start: from,
+            end: to,
+          },
+          kqlQuery: {
+            filterQuery: {
+              kuery: {
+                kind: (timeline.kqlQuery?.filterQuery?.kuery?.kind ??
+                  'kuery') as KueryFilterQueryKind,
+                expression: query,
+              },
+              serializedQuery: convertKueryToElasticSearchQuery(query),
+            },
+            filterQueryDraft: {
+              kind: (timeline.kqlQuery?.filterQuery?.kuery?.kind ??
+                'kuery') as KueryFilterQueryKind,
```

Comment:
This `as cast` isn't needed

Reply:
Removing them also cleans up the imports more at the top where you can delete all the type imports which is pretty awesome.